src/templates.rs

```diff
@@ -103,125 +107,178 @@ pub fn render_template_file_tree<'a, T: Serialize>(
-            )
-            .unwrap();
-            let if_regex = Regex::new(
-                r"(?P<c>(.)*)/\{\{#if (?P<b>([^\{\}])*)\}\}(?P<a>(.)*)\{\{/if\}\}.hbs\z",
-            )
-            .unwrap();
-
-            if re.is_match(path.to_str().unwrap()) {
-                let path_prefix = re.replace(path.to_str().unwrap(), "${c}");
-                let path_prefix = h.render_template(path_prefix.to_string().as_str(), data)?;
-
-                let new_path_suffix =
-                    re.replace(path.to_str().unwrap(), "{{#each ${b} }}${a}.hbs{{/each}}");
-
-                let all_paths = h.render_template(new_path_suffix.to_string().as_str(), data)?;
-
-                let files_to_create: Vec<String> = all_paths
-                    .split(".hbs")
-                    .map(|s| s.to_string())
-                    .filter(|s| !s.is_empty())
-                    .collect();
-
-                if files_to_create.len() > 0 {
-                    let delimiter = "\n----END_OF_FILE_DELIMITER----\n";
-
-                    let each_if_re = Regex::new(
-                    r"(?P<c>(.)*)/\{\{#each (?P<b>([^\{\}])*)\}\}\{\{#if (?P<d>([^\{\}])*)\}\}(?P<a>(.)*)\{\{/if\}\}\{\{/each\}\}.hbs\z",
-                )
-                .unwrap();
-                    let b = re.replace(path.to_str().unwrap(), "${b}");
-                    let new_all_contents = match each_if_re.is_match(path.to_str().unwrap()) {
-                        true => {
-                            let d = each_if_re.replace(path.to_str().unwrap(), "${d}");
-                            format!(
-                                "{{{{#each {} }}}}{{{{#if {} }}}}\n{}{}{{{{/if}}}}{{{{/each}}}}",
-                                b, d, contents, delimiter
-                            )
-                        }
-
-                        false => format!(
-                            "{{{{#each {} }}}}\n{}{}{{{{/each}}}}",
-                            b, contents, delimiter
-                        ),
-                    };
-                    let new_contents = render_template_file(
-                        &h,
-                        existing_app_file_tree,
-                        &path,
-                        &new_all_contents,
-                        &value,
-                    )?;
-                    let new_contents_split: Vec<String> = new_contents
-                        .split(delimiter)
-                        .into_iter()
-                        .map(|s| s.to_string())
-                        .collect();
-
-                    for (i, f) in files_to_create.into_iter().enumerate() {
-                        let target_path = PathBuf::from(path_prefix.clone()).join(f);
-
-                        transformed_templates
-                            .insert(target_path, Some(new_contents_split[i].clone()));
+            let each_regex = Regex::new(EACH_TEMPLATE_REGEX).context("EACH_TEMPLATE_REGEX is invalid")?;
```

Comment:
This is better with constants but I'd go a step further and try to avoid recreating the `Regex`. Especially in a loop